CMakeLists.txt

```diff
@@ -532,7 +534,7 @@ else()
   set(BOOST_SOURCE_DIR "${PROJECT_SOURCE_DIR}/src/boost")
   set(BOOST_PREFIX "${PROJECT_BINARY_DIR}/boost")
   set(BOOST_BUILD "${PROJECT_BINARY_DIR}/boost-build")
-  string(REPLACE ";" "," BOOST_WITH_LIBS "${BOOST_COMPONENTS}")
+  string(REPLACE ";" "," BOOST_WITH_LIBS "${BOOST_COMPONENTS} ${BOOST_HEADER_COMPONENTS}")
```

Comment:
i think we need to `list(APPEND BOOST_COMPONENTS ${BOOST_HEADER_COMPONENTS}")` before doing the REPLACE.